modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/auth/ConditionalAuthenticationTestCase.java

```diff
@@ -147,6 +151,40 @@ public void atEnd() throws Exception {
     @Test(groups = "wso2.is", description = "Check conditional authentication flow.")
     public void testConditionalAuthentication() throws Exception {
 
+        LoginToPrimaryIS();
+        /* Here if the client is redirected to the secondary IS, it indicates that the conditional authentication steps
+         has been successfully completed. */
+        assertTrue(response.getFirstHeader("location").getValue().contains(SECONDARY_IS_SAMLSSO_URL),
+                "Failed to follow the conditional authentication steps.");
+        EntityUtils.consume(response.getEntity());
+    }
+
+    @Test(groups = "wso2.is", description = "Check conditional authentication flow based on HTTP Context.")
+    public void testConditionalAuthenticationUsingHTTPContext() throws Exception {
```

Comment:
we better rename this method to 
testConditionalAuthenticationUsingHTTPCookie
As there are more to test for HTTPContext, e.g. Header, IP